include/termdb.hpp

```diff
@@ -1382,6 +1403,223 @@ std::string TermDb::parser(const std::string &s, param p1, param p2, param p3,
     return incorrectString ? "" : result;
 }
 
+#elif defined(OS_WIN)
+
+class TermDb {
+private:
+    std::string name;
+    bool isValidState    = false;
+    bool isLessThanWin10 = false;
+    HANDLE consoleHandle = nullptr;
+
+    void _emulate_clear() const noexcept;
+    void _emulate_cup(const short) const noexcept;
+    void _emulate_cud(const short) const noexcept;
+    void _emulate_cuf(const short) const noexcept;
+    void _emulate_cub(const short) const noexcept;
+    void _emulate_cha(const short) const noexcept;
+    void _emulate_vpa(const short) const noexcept;
+    void _emulate_cnorm() const noexcept;
+    void _emulate_civis() const noexcept;
+    void _emulate_cpl(const short) const noexcept;
+    void _emulate_cnl(const short) const noexcept;
+
+    std::string do_clear() const noexcept;
+
+public:
+    TermDb() = default;
+
+    TermDb(const std::string &, std::string) = delete;
+    bool parse(const std::string, std::string);
+
+    explicit operator bool() const noexcept { return isValidState; }
+    std::string getName() const { return name; }
+
+    nonstd::optional<uint16_t> get(tdb::num _n) const { return 0; }
+    bool get(tdb::bin _b) const noexcept { return false; }
+
+    std::string parser(const std::string &s, param p1, param p2, param p3,
+                       param p4, param p5, param p6, param p7, param p8,
+                       param p9) const = delete;
+
+    std::string get(tdb::str _s, param p1 = 0l, param p2 = 0l, param p3 = 0l,
+                    param p4 = 0l, param p5 = 0l, param p6 = 0l, param p7 = 0l,
+                    param p8 = 0l, param p9 = 0l) const
+    {
+
+        if (!isValidState) {
+            return "";
+        }
+
+        switch (_s) {
+            case tdb::str::clear_screen: return do_clear(); break;
+        }
+    }
+};
+
+bool TermDb::parse(const std::string _name = "", std::string _path = "")
+{
+    consoleHandle = GetStdHandle(STD_OUTPUT_HANDLE);
+    if (consoleHandle == INVALID_HANDLE_VALUE) {
+        isValidState = false;
+        return isValidState;
+    } else {
+        isValidState = true;
+    }
+
+    isLessThanWin10 = !IsWindows10OrGreater();
+    name            = "cmd.exe";
```

Comment:
can remove name as a member variable itself and `return "cmd.exe"` directly from `getName()`